### PR TITLE
Speed up Naga's `cargo xtask validate wgsl` from 12s to 0.8s

### PR DESCRIPTION
**Pull request by jimblandy**
Implement a `--bulk-validate` option in the Naga CLI, and use it in Naga's `cargo xtask validate wgsl` to validate all WGSL files in a single execution. This reduces real time from 12s to 0.8s.

**Checklist**

- [X] Run `cargo fmt`.
- [X] Run `cargo clippy`.
- [X] Run `cargo xtask test` to run tests.
- [x] Add change to `CHANGELOG.md`. See simple instructions inside file.

### CODE REVIEW

**Comment by jimblandy** [comment]
I have the feeling this code isn't the greatest. I wanted to get it out there and then just do whatever people tell me to do.

File: naga-cli/src/bin/naga.rs
Comment (ErichDonGubler) [inline naga-cli/src/bin/naga.rs:676]: nitpick(non-blocking): I'd combine these into a single lock-'n'-write with `eprintln!("Error validating {}:\n{}", …)`.

File: naga-cli/src/bin/naga.rs
Comment (ErichDonGubler) [inline naga-cli/src/bin/naga.rs:254]: nitpick(non-blocking): This move of init. for the `params` binding doesn't seem necessary? I wouldn't change it now, since both author and reviewer now understand the code, but it tooks me a bit to figure out that this movement doesn't meaningfully change anything.

File: naga/xtask/src/main.rs
Comment (ErichDonGubler) [inline naga/xtask/src/main.rs:200]: praise: LGTM here! I don't see anything I'd like to change here; it's pretty straightforward on the `xtask` end. 🙂

**Review by ErichDonGubler — APPROVED** [review]
@jimblandy: I don't think I have enough context to authoritatively respond to your [implicit plea for validation of code quality](https://github.com/gfx-rs/wgpu/pull/4871#issuecomment-1853056015) WRT the Naga CLI experience, so I'll ask: do you feel like it concretely comes up short in some way, or doesn't fit well into the existing experience? I didn't see anything obvious (as a newcomer).

**Comment by ErichDonGubler** [comment]
Looks like we need a @gfx-rs/naga review here still. Do you think it's worth waiting for @teoxoy, @jimblandy?

**Comment by jimblandy** [comment]
> Looks like we need a @gfx-rs/naga review here still. Do you think it's worth waiting for @teoxoy, @jimblandy?

Maybe for this it's okay.

**Comment by jimblandy** [comment]
Merging without review from code owners, as this is a pretty simple change that doesn't affect Naga itself.

File: naga-cli/src/bin/naga.rs
Comment (jimblandy) [reply naga-cli/src/bin/naga.rs:676]: That's a good point. I think it's a bit more legible this way, though.

Comment (ErichDonGubler) [reply naga-cli/src/bin/naga.rs:676]: Fun fact (not intended to dissuade you): you can also use the `concat!(...)` macro, too:

```rust
eprint!(
    concat!(
        "Line 1: {}\n",
        "Line 2: {}\n"
    ),
    arg1,
    arg2
);
```

**Pull request merged**
